### DISCUSSION TITLE
TS2345: Argument of type 'unknown' is not assignable to parameter of type 'string'

### DISCUSSION THREAD

**Discussion by ptmkenny** [react-hook-form/react-hook-form#4377] [Q&A]
When I added `defaultValues` to `useForm`, I started getting this error:

`TS2345: Argument of type 'unknown' is not assignable to parameter of type 'string'.`

Example code that works:

```
  const {
    register, handleSubmit, watch, setValue, control,
  } = useForm();

  const selectedGroup = watch('group');
...
      {checkIfAddGroup(selectedGroup) && (
        <>
          <ShowGroup />
        </>
      )}
```

But when I set `defaultValues` for useForm, I get the TypeScript error for the `checkIfAddGroup(selectedGroup)` line.

```
  const {
    register, handleSubmit, watch, setValue, control,
  } = useForm({
    // We need to set these defaults to avoid "undefined" if the user does not edit the field.
    // (CKEditor will returned undefined if the field isn't modified.)
    defaultValues: {
      question: passedDefaultValues.question,
      answer: passedDefaultValues.answer,
    },
  });

  const selectedGroup = watch('group');
...
      {checkIfAddGroup(selectedGroup) && (
        <>
          <ShowGroup />
        </>
      )}
```

I'm still learning TypeScript so I may be missing something obvious; I would appreciate any insight you could provide.

I don't understand why changing the default values affects another part of the form.

**Comment by jorisre** [comment]
@ptmkenny Can you provide an example in www.codesandbox.io please ?
